buildSrc/src/main/java/org/elasticsearch/gradle/release/ReleaseToolsPlugin.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.gradle.release;
+
+import org.elasticsearch.gradle.Version;
+import org.elasticsearch.gradle.VersionProperties;
+import org.elasticsearch.gradle.internal.precommit.ValidateYamlAgainstSchemaTask;
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.provider.Provider;
+import org.gradle.api.tasks.util.PatternSet;
+
+import java.io.File;
+
+/**
+ * This plugin defines tasks related to releasing Elasticsearch.
+ */
+public class ReleaseToolsPlugin implements Plugin<Project> {
+
+    @Override
+    public void apply(Project project) {
+        final Provider<ValidateYamlAgainstSchemaTask> validateChangelogs = project.getTasks()
+            .register("validateChangelogs", ValidateYamlAgainstSchemaTask.class, task -> {
+                task.setGroup("Documentation");
+                task.setDescription("Validates that all the changelog YAML files are well-formed");
+
+                task.setInputFiles(
+                    project.getLayout()
```

Comment:
Instead of relying on project.getLayout() we can inject `ProjectLayout` via the constructor here. This reads cleaner imo and also helps us later on when we wanna migrate to --configuration-cache were referecing the mutable project instance is discouraged from several places.